### ISSUE TITLE
Release strategy for muffin 4.2

### ISSUE THREAD

**Issue by clefebvre** [linuxmint/cinnamon#8602]
Builds
--------

[4.0.zip](https://github.com/linuxmint/cinnamon/files/3007779/4.0.zip)
[4.0-maintenance.zip](https://github.com/linuxmint/cinnamon/files/3004092/4.0-maintenance.zip)
[iter-10.zip](https://github.com/linuxmint/cinnamon/files/3256875/iter-10.zip)
[iter-20.zip](https://github.com/linuxmint/cinnamon/files/3257142/iter-20.zip)
[revert-20.zip](https://github.com/linuxmint/cinnamon/files/3258067/revert-20.zip)
[reimplement-20.zip](https://github.com/linuxmint/cinnamon/files/3262648/reimplement-20.zip)
[master-5.6.zip](https://github.com/linuxmint/cinnamon/files/3258330/master-5.6.zip)

Note: 

- iter-10 consists in the master branches of muffin/cinnamon just before we started working on rect caching and obstructed windows optimizations.
- iter-20 consists in iter-10 plus all commits backported from current master, except for commits related to rect caching and obstructed windows, or commits which don't apply cleanly without these.
- revert-20 consists in current master with rect caching and obstructed windows optimizations removed using reverts. It's the same as iter-20, but using reverts from master, instead of cherry-picks from iter-10/4.0.
- reimplement-20 consists in revert-20 with a reimplementation of the commits which didn't apply cleanly and had to be reverted in revert-20. It's pretty much the same as master, but without rect caching and obstructed windows optimizations.

Plan A: Fix remaining issues and release master
-------------------------------------------------------------

Issue 1: Pkexec Xed
--------------------------

Open a terminal and type "pkexec xed", observe how the content of the dialog doesn't map correctly. The dialog first shows the titlebar and then the content looks like it's sliding downwards into position.

This is observable with "./test" in the mintupdate project as well. The mintupdate window first maps in its top-left corner, then the content arrives.

| Introduced by | Fixed by |
|     :---:             |     :---:     |
|  https://github.com/linuxmint/muffin/pull/410 | https://github.com/linuxmint/muffin/pull/453 (partly) |

Issue 2: Screensaver doesn't react to input
-------------------------------------------------------

Issue 3: Window close animation
------------------------------------------

Set close animation to 1000ms, close a window, observe broken animation.

Issue 4: after restarting Cinnamon fullscreen windows not being positioned correctly
----------------------------------------------------------------------------------------------------------
when not on the primary workspace

Issue 5: windows disappearing sometimes when switching workspaces
-----------------------------------------------------------------------------------------
mousing over the window list until the preview shows fixes that

Plan B: Release without rect caching and obstructed windows optimizations
-----------------------------------------------------------------------------------------------

Most of the issues we've been struggling with originate from these two optimizations or the refactors that came with them. The idea is to keep everything else (there are plenty of other optimizations in muffin and cinnamon master since 4.0) by doing this:

- Revert rect caching / obstructed wins optimizations in muffin: https://github.com/linuxmint/muffin/pull/515
- Revert rect caching / obstructed wins optimizations in cinnamon: https://github.com/linuxmint/cinnamon/pull/8606
- Reimplement "Make the sync method configurable"
- Reimplement a large part of "layout: Misc. clean up"
- Reimplement "Replace some GLib wrappers"
- Reimplement "Replace some deprecated GDK methods"
- Reimplement "Expose the sync method to preferences"
- Reimplement "Fix unredirection for wine games, mpv"
- Reimplement "Add MetaPrefsState struct for retrieving the prefs state without function invocation"
- Release as 4.2
- Turn rect caching + obstructed windows (+ fixes) as a PR for 4.4 (we can do this easily by squashing the reverts together in the revert-20 PRs and reverting the revert in the new PR).
